etna/models/sarimax.py

```diff
@@ -0,0 +1,426 @@
+            mle_regression=self.mle_regression,
+            simple_differencing=self.simple_differencing,
+            enforce_stationarity=self.enforce_stationarity,
+            enforce_invertibility=self.enforce_invertibility,
+            hamilton_representation=self.hamilton_representation,
+            concentrate_scale=self.concentrate_scale,
+            trend_offset=self.trend_offset,
+            use_exact_diffuse=self.use_exact_diffuse,
+            dates=self.dates,
+            freq=self.freq,
+            missing=self.missing,
+            validate_specification=self.validate_specification,
+            **self.kwargs,
+        )
+        # expect every params but last to be near 0
+        start_params = [0, 0, 0, 0] + addition_to_params + [1]
+        self._result = self._model.fit(start_params=start_params, disp=False)
+        return self
+
+    def predict(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Compute predictions from a SARIMAX model.
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            Features dataframe
+
+        Returns
+        -------
+        y_pred: pd.DataFrame
+            DataFrame with predictions
+        """
+        horizon = len(df)
+        self._check_df(df, horizon)
+
+        categorical_cols = df.select_dtypes(include=["category"]).columns.tolist()
+        try:
+            df.loc[:, categorical_cols] = df[categorical_cols].astype(int)
+        except ValueError:
+            raise ValueError(
+                f"Categorical columns {categorical_cols} can not been converted to int.\n "
+                "Try to encode this columns manually."
+            )
+
+        exog_future = self._select_regressors(df)
+        y_pred = self._result.predict(
+            start=df["timestamp"].min(), end=df["timestamp"].max(), dynamic=True, exog=exog_future
+        )
+        return y_pred.reset_index(drop=True, inplace=False)
+
+    def _check_df(self, df: pd.DataFrame, horizon: Optional[int] = None):
+        column_to_drop = [
+            col for col in df.columns if not col.startswith("regressor") and col not in ["target", "timestamp"]
+        ]
+        regressor_columns = [col for col in df.columns if col.startswith("regressor")]
+        if column_to_drop:
+            warnings.warn(
+                message=f"SARIMAX model does not work with exogenous features (features unknown in future).\n "
+                f"{column_to_drop} will be dropped"
+            )
+        if horizon:
+            short_regressors = [regressor for regressor in regressor_columns if df[regressor].count() < horizon]
+            if short_regressors:
+                raise ValueError(
+                    f"Regrssors {short_regressors} are too short for chosen horizon value.\n "
+                    "Try lower horizon value, or drop this regressors."
+                )
+
+    def _select_regressors(self, df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
as soon as `_select_regressors` can return `None` it should be
```suggestion
    def _select_regressors(self, df: pd.DataFrame) -> Optional[pd.DataFrame]:
```